### DISCUSSION TITLE
Error: Could not find a relationship between 'table1' and 'table2' in the schema cache

### DISCUSSION THREAD

**Discussion by TimDev9492** [supabase/supabase#11914] [Questions] [answered]
I have these two tables set up like this:
```sql
create table schooldata.schools (
  "school_id" integer not null,
  "school_name" text not null,
  "city_name" text,
  constraint "schools_pkey" primary key ("school_id")
);

create table schooldata.schoolyears (
  "id" serial not null,
  "year_name" text not null,
  "school_id" integer not null,
  constraint "schoolyears_pkey" primary key ("id")
);

alter table schooldata.schoolyears
add constraint "schoolyears_schoolid_fkey" foreign key ("school_id") references schooldata.schoolyears("school_id") on delete restrict on update cascade;
```
and the 'schoolyears' table looks like this
![image](https://user-images.githubusercontent.com/67952738/214573136-8de28b9a-c770-49e2-b6a1-acfd58c36d99.png)
![image](https://user-images.githubusercontent.com/67952738/214573196-75ad1863-c33b-4842-a3fc-c8915e36f6d5.png)
...which suggests that supabase recognises some kind of relation between the two.

However, when I run the following query:
```js
await supabaseSchooldata
    .from('schools')
    .select(`*, schoolyears(*)`)
    .order('school_name');
```
I get the error `Could not find a relationship between 'schools' and 'schoolyears' in the schema cache`.
I already tried reloading the schema cache with the following sql command:
```sql
NOTIFY pgrst, 'reload schema';
```
but that didn't help either. Am I missing something?

**Comment by GaryAustin1** [comment, marked as answer]
You seem to be using a user generated schema.  Can you access each table thru the API by itself?   I assume you are setting the schema in the createClient call.  Could be a grant problem on one of the tables, depending on how you set up grants on the schema.

RLS can cause this error if one table is not accessible to the user.

**Reply by GaryAustin1** [reply]
Also normally the key pointed to by a foreign key needs to be primary or have an unique constraint.  I don't see that in your table code.

**Reply by TimDev9492** [reply]
I am using a user generated schema. I managed to fix the issue. I wasn't passing the `{ db: { schema: 'schooldata' } }` options when creating the supabase client.
